### PR TITLE
💻 Aggregating Reviews on Resources

### LINKED ISSUE #165: Aggregating rating
Provide an aggregate rating on each resource page.

image

### PR DESCRIPTION
**Pull request by a22yuen**
# Purpose
Closes #165

# Screenshots
image

- Display aggregated reviews within each resource and on the card when in grid form for overviewing all resources

### CODE REVIEW

File: client/components/common/Functions.tsx [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const capitalizeFirstLetter = (str: string): string => {
```
Comment (jeffzh4ng) [inline client/components/common/Functions.tsx:1]: `/common/Functions` is somewhat of a weird name. Something like `/common/utils` might be better?

File: client/components/resource/ResourceCard.tsx [hunk 2]
```diff
@@ -100,6 +131,8 @@ export const ResourceCard = ({ Resource }: ResourceCardProps) => {
     router.push({ pathname: `/${resourceSlug}/${id}` })
   }
 
+  console.log(resourceLookup[resourceSlug]['criteria'])
```
Comment (jeffzh4ng) [inline client/components/resource/ResourceCard.tsx:134]:
```suggestion
```

File: client/components/resource/ResourceCard.tsx [hunk 3]
```diff
@@ -113,6 +146,21 @@ export const ResourceCard = ({ Resource }: ResourceCardProps) => {
           <TagGroup>{tags.map(ResourceCardTag)}</TagGroup>
         </TagRow> */}
       </ResourceCardContent>
+      <StatisticsRow>
+        {resourceLookup[resourceSlug]['criteria'].map((criteria) => (
```
Comment (jeffzh4ng) [inline client/components/resource/ResourceCard.tsx:150]:
```suggestion
        {resourceLookup[resourceSlug]['criteria'].map((criterion) => (
```

**Review by jeffzh4ng — CHANGES_REQUESTED** [review]
Amazing job guys, the preview looks fucking great 😍

File: client/components/resource-details/ResourceInfo.tsx [hunk 4]
```diff
@@ -1,73 +1,160 @@
...
 
       <Name>{name}</Name>
 
-      <Metadata>
-        <MemberCount>200-300 students</MemberCount>
-      </Metadata>
+      <RatingDescription>
+        <Description>
+          <Metadata>
+            <MemberCount>200-300 students</MemberCount>
```
Comment (xujustinj) [inline client/components/resource-details/ResourceInfo.tsx:142]: TODO (not for this PR)

File: client/styles/typography.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import styled from 'styled-components'
 import { colours } from './colours'
-import { width, smallerThan } from './device'
+import { width, smallerThan, largerThan } from './device'
```
Comment (xujustinj) [inline client/styles/typography.ts:3]:
```suggestion
import { width, smallerThan } from './device'
```

**Review by xujustinj — APPROVED** [review]

File: client/components/resource/ResourceCard.tsx [hunk 6]
```diff
@@ -92,6 +90,50 @@ interface ResourceCardProps {
   Resource: Resource
 }
 
+const StatisticsRow = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  // display: grid;
+  // grid-template-columns: repeat(auto-fit, minmax(100px, 33%));
```
Comment (jeffzh4ng) [inline client/components/resource/ResourceCard.tsx:97]:
```suggestion
```

**Review by jeffzh4ng — APPROVED** [review]
🔥

**Pull request merged**